cirq-google/cirq_google/api/v2/results_test.py

```diff
@@ -175,17 +188,71 @@ def test_results_to_proto():
             )
 
 
-def test_results_to_proto_sweep_repetitions():
-    measurements = [v2.MeasureInfo('foo', [q(0, 0)], slot=0, invert_mask=[False], tags=[])]
+def test_results_to_proto_repeated_keys():
+    measurements = [
+        v2.MeasureInfo('foo', [q(0, 0)], instances=2, invert_mask=[False], tags=[]),
+        v2.MeasureInfo('bar', [q(0, 0)], instances=1, invert_mask=[False], tags=[]),
+    ]
     trial_results = [
         [
             cirq.ResultDict(
                 params=cirq.ParamResolver({'i': 0}),
-                measurements={'foo': np.array([[0]], dtype=bool)},
+                records={
+                    'foo': np.array([[[0], [0]], [[0], [1]], [[0], [0]], [[0], [1]]], dtype=bool),
+                    'bar': np.array([[[0]], [[1]], [[1]], [[0]]], dtype=bool),
+                },
+            ),
+            cirq.ResultDict(
+                params=cirq.ParamResolver({'i': 1}),
+                records={
+                    'foo': np.array([[[0], [0]], [[0], [1]], [[0], [1]], [[0], [0]]], dtype=bool),
+                    'bar': np.array([[[0]], [[1]], [[1]], [[0]]], dtype=bool),
+                },
+            ),
+        ],
+        [
+            cirq.ResultDict(
+                params=cirq.ParamResolver({'i': 0}),
+                records={
+                    'foo': np.array([[[0], [0]], [[0], [1]], [[0], [0]], [[0], [1]]], dtype=bool),
+                    'bar': np.array([[[0]], [[1]], [[1]], [[0]]], dtype=bool),
+                },
+            ),
+            cirq.ResultDict(
+                params=cirq.ParamResolver({'i': 1}),
+                records={
+                    'foo': np.array([[[0], [0]], [[0], [1]], [[0], [1]], [[0], [0]]], dtype=bool),
+                    'bar': np.array([[[0]], [[1]], [[1]], [[0]]], dtype=bool),
+                },
+            ),
+        ],
+    ]
+    proto = v2.results_to_proto(trial_results, measurements)
+    assert isinstance(proto, v2.result_pb2.Result)
+    assert len(proto.sweep_results) == 2
+    deserialized = v2.results_from_proto(proto, measurements)
+    assert len(deserialized) == 2
```

Comment:
Prefer to test behavior independently rather than rolling many separate assertions into one monolithic test.

go/unit-testing-practices#behavior-testing-examples